### ISSUE TITLE
Add Telegram as a social login provider

### ISSUE THREAD

**Issue by trashcluster** [goauthentik/authentik#2232]
Telegram offers an authentication mechanism for websites, i've seen some websites that did this --> https///portal.bcc.no. The login page is Auth0 so integrating it is possible

https://telegram.org/blog/login
https://core.telegram.org/widgets/login

I've searched without luck for a keycloak plugin for this function but nobody seems to have investigated it but now that i've come across Authentik project it seems way more suited for it.

**Comment by Ceralor** [comment]
Contributing to interest in this. It'd be useful under numerous circumstances, including the ability to filter based on groups someone is a member of.

**Comment by Didza** [comment]
This feature would be greatly appreciated

**Comment by mahdikiani** [comment]
I think this feature is more needed now

**Comment by prostoprojekt** [comment]
Yep, that would be great to see Telegram auth in nearest releases!

**Comment by Didza** [comment]
We would really appreciate this feature.

**Comment by Mrs-Feathers** [comment]
yes! i wonder how easy it would be to generate an automatic login URL via api... a telegram bot could request it for the user who's account is linked and then the bot would send a login url for the user to click and done

**Comment by sysadminpower2019** [comment]
any updates on this?

**Comment by Ceralor** [comment]
@Mrs-Feathers I may give it a try using a variation on the Email stage, storing the user's information in attributes.

**Comment by sepffuzzball** [comment]
I'd upvote this as well!

**Comment by DRSchlaubi** [comment]
The big issue is that despite the URL using oauth.telegram.org not having a lot of oauth in it, therefore, most tools like authentik don't support it, the way how portal.bcc.no seems to have done it, is by implementing their own login provider, which converts telegram to something auth0 supports.

I have done a very hack poc of this [here](https://github.com/DRSchlaubi/telegram-oauth) which is compatible with authentik until we have someting permanent, simmilar to this comment in the steam issue https://github.com/goauthentik/authentik/issues/2282#issuecomment-1574993069

**Comment by anataliocs** [comment]
I'd appreciate this feature as well!!!

**Comment by ARHAEEM** [comment]
Hope to see this soon.

**Comment by joshmarom** [comment]
Would love that too 😌

**Comment by Grommash9** [comment]
Would love that too 😌

**Comment by jayp89** [comment]
Give us telegram

**Comment by xfo** [comment]
Would love that too 😌

**Comment by Brukkil** [comment]
+1

**Comment by dewi-tik** [comment]
Work is underway to add telegram (see linked PR). Can't confirm release at the moment.

**Comment by OctoNezd** [comment]
Hi! I am trying to use Telegram auth on ghcr.io/goauthentik/dev-server:gh-next. I get this error:
```
{"auth_via": "unauthenticated", "domain_url": "auth.octonezd.me", "event": "Refusing to use none property", "host": "auth.octonezd.me", "identifier": 174781687, "level": "warning", "logger": "authentik.core.sources.matcher", "pid": 66, "property": "email", "request_id": "f4982a08647341c7b884a1d1eb79da05", "schema_name": "public", "source": "<TelegramSource: Telegram>", "timestamp": "2025-10-10T13:32:54.284891"}
```

I set source to use username and not email, but it didnt help. Is feature incomplete right now?

**Comment by atereshkin** [comment]
> Hi! I am trying to use Telegram auth on ghcr.io/goauthentik/dev-server:gh-next. I get this error:
> 
> ```
> {"auth_via": "unauthenticated", "domain_url": "auth.octonezd.me", "event": "Refusing to use none property", "host": "auth.octonezd.me", "identifier": 174781687, "level": "warning", "logger": "authentik.core.sources.matcher", "pid": 66, "property": "email", "request_id": "f4982a08647341c7b884a1d1eb79da05", "schema_name": "public", "source": "<TelegramSource: Telegram>", "timestamp": "2025-10-10T13:32:54.284891"}
> ```
> 
> I set source to use username and not email, but it didnt help. Is feature incomplete right now?

@OctoNezd could you share a bit more details about your setup? I've just tried a vanilla beta install and logging in via Telegram worked for me.
 Are you using some specific property mappings along with the Telegram source? What other configuration have you done beyond what's described in **https://github.com/goauthentik/authentik/blob/main/website/docs/users-sources/sources/social-logins/telegram/index.md**?

**Comment by OctoNezd** [comment]
I didnt do any property mappings. I didnt manage to find doc myself, thank you for linking it - I will try it this weekend

**Comment by Deniom3** [comment]
Hello
Could you tell me what the property mapping should look like to ensure authorization works for existing users?
I created the mapping:

return {
     "telegram_id": info.get("id")
}

Added a property to an existing user telegram_id  and filled it with the correct value.
But when I try to log in via Telegram, it just creates a new user instead of using the existing one. which has exactly the same properties as I indicated, only it is located in a different folder and has a different name.
 What am I doing wrong?

**Comment by atereshkin** [comment]
@Deniom3 Unfortunately your options are fairly limited as far as I can tell. Whether and how the source user will be linked to an existing user is determined by the source's _User matching mode_.  The default "link users on unique identifier" mode will always create a new user if this Telegram user hasn't logged in before. In order to link to an existing one you can use the "Link  to a user with identical username" mode and use a property mapping that will match telegram ids to usernames like this:
```
if info["id"] == <specific telegram id>:
  return {"username": "<existing user's username>"}
return {}
```
One downside is that you'll have to hard-code all the known users in this property mapping. But the worse downside is that it would allow anyone log in as any user of your installation by registering the corresponding username on Telegram. You can work around that in a slightly hacky way by using email-based matching mode because the Telegram source doesn't set email address so you wouldn't run into the risk of accidentally matching an existing user.

**Comment by jelle619** [comment]
Hey everyone. Great to see this released! However, it does not look like it is possible to link Telegram accounts to existing accounts. In connected services in the account settings, instead of showing a connect button, it simply shows a dash. Anyone else who has this issue?

**Comment by Nikita-Silent** [comment]
Hi, guys! I've the same issue as @jelle619. Can anyone help us?

**Comment by atereshkin** [comment]
@jelle619 @Nikita-Silent thanks for pointing this out! On it. This will be tracked in #18467

**Comment by DRSchlaubi** [comment]
Telegram has official support for OIDC now: https://core.telegram.org/bots/telegram-login#openid-connect
